Client/Frontend/Widgets/AuralProgressBar.swift

```diff
@@ -185,4 +164,35 @@ class AuralProgressBar {
             }
         }
     }
+
+    init() {
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: Selector("handleAudioSessionInterruptionNotification:"), name: AVAudioSessionInterruptionNotification, object: nil)
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: AVAudioSessionInterruptionNotification, object: nil)
+    }
+
+    @objc func handleAudioSessionInterruptionNotification(notification: NSNotification) {
+        if let interruptionTypeValue = notification.userInfo?[AVAudioSessionInterruptionTypeKey] as? UInt {
+            if let interruptionType = AVAudioSessionInterruptionType(rawValue: interruptionTypeValue) {
+                switch interruptionType {
+                case .Began:
+                    ui.stop()
+                case .Ended:
+                    if let interruptionOptionValue = notification.userInfo?[AVAudioSessionInterruptionOptionKey] as? UInt {
+                        let interruptionOption = AVAudioSessionInterruptionOptions(rawValue: interruptionOptionValue)
+                        if interruptionOption == .ShouldResume {
+                            if !hidden {
+                                if let progress = progress {
```

Comment:
Can we merge this into the `if let` above to reduce nesting? `let progress = progress where !hidden && interruptionOption == .ShouldResume`, perhaps?